test/origin_simulator/multi_route_origin_simulator_test.exs

```diff
@@ -0,0 +1,40 @@
+defmodule MultiRouteOriginSimulatorTest do
+  use ExUnit.Case
+  use Plug.Test
+
+  setup do
+    OriginSimulator.Simulation.restart()
+    Process.sleep(10)
+  end
+
+  describe "GET pages" do
+    test "for a wildcard route" do
+      payload = [
+        %{
+          "route" => "/*",
+          "body" => "I am ok",
+          "stages" => [%{"at" => 0, "status" => 200, "latency" => 0}]
+        }
+      ]
+
+      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
+      Process.sleep(20)
+
+      path = "/any_path"
+      conn = conn(:get, path)
+      conn = OriginSimulator.call(conn, [])
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert conn.resp_body == "I am ok"
+
+      path = "/another_path"
+      conn = conn(:get, path)
+      conn = OriginSimulator.call(conn, [])
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert conn.resp_body == "I am ok"
```

Comment:
Would it be better to split these two sets of asserts into two tests? They do look very similar but I'm wondering if there is a reason why we have two sets? If its arbitrary then one set would be ok I think and less ambiguous.

Reply:
To demonstrate that `/*` works for any-many routes. Two paths should be the minimum testing criteria. This should be refactored in the same way as the single-route tests (from the next PR), cf. the following "arbitrary wildcard" test case :

```ex
    test "for '/*' route", %{payload: payload} do
      payload = Map.put(payload, "route", "/*")
      conn(:post, "/#{admin_domain()}/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
      Process.sleep(20)

      conn(:get, "/news/uk-politics-51287430")
      |> OriginSimulator.call([])
      |> assert_status_body(200, body_mock())

      conn(:get, "/sport/tennis/51291122")
      |> OriginSimulator.call([])
      |> assert_status_body(200, body_mock())
    end

    test "for arbitrary wildcard route", %{payload: payload} do
      payload = Map.put(payload, "route", "/news*")
      conn(:post, "/#{admin_domain()}/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
      Process.sleep(20)

      conn(:get, "/news/uk-politics-51287430")
      |> OriginSimulator.call([])
      |> assert_status_body(200, body_mock())
    end
```

Actually need to improve the test coverage for multi-route simulation!

Reply:
I guess you could say that two is still arbitrary as there is no guarantee the code isn't hardcoded to handle the two paths. I think in the original code here that testing the route `/any_path` or `/some-wildcard-path` would be better than `/news/uk-politics-51287430` as that helps describe the intent of it being a wildcard path.